BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BTConnectToThread.java

```diff
@@ -4,32 +4,46 @@
 
 import android.bluetooth.BluetoothDevice;
 import android.bluetooth.BluetoothSocket;
-import android.util.Log;
-
+import android.os.CountDownTimer;
+import android.os.Handler;
+import android.os.Message;
 import java.io.IOException;
 import java.util.UUID;
 
 /**
  * Created by juksilve on 12.3.2015.
  */
-public class BTConnectToThread extends Thread {
+class BTConnectToThread extends Thread{
 
     public interface  BtConnectToCallback{
-        public void Connected(BluetoothSocket socket,String peerId,String peerName,String peerAddress);
-        public void ConnectionFailed(String reason,String peerId,String peerName,String peerAddress);
+        void Connected(BluetoothSocket socket,String peerId,String peerName,String peerAddress);
+        void ConnectionFailed(String reason,String peerId,String peerName,String peerAddress);
     }
 
-    private BtConnectToCallback callback;
+    private BTHandShakeSocketTread mBTHandShakeSocketTread = null;
+    private String mInstanceString = "";
+    private final BtConnectToCallback callback;
     private final BluetoothSocket mSocket;
-    String mPeerId;
-    String mPeerName;
-    String mPeerAddress;
+    private final String mPeerId;
+    private final String mPeerName;
+    private final String mPeerAddress;
+
+    private final CountDownTimer HandShakeTimeOutTimer = new CountDownTimer(4000, 1000) {
+        public void onTick(long millisUntilFinished) {
+            // not using
+        }
+        public void onFinish() {
+            HandShakeFailed("TimeOut");
+        }
+    };
 
-    public BTConnectToThread(BtConnectToCallback Callback, BluetoothDevice device, UUID BtUUID, String peerId,String peerName, String peerAddress) {
+    public BTConnectToThread(BtConnectToCallback Callback, BluetoothDevice device, UUID BtUUID, String peerId,String peerName, String peerAddress, String InstanceString) {
         callback = Callback;
         mPeerId = peerId;
         mPeerName = peerName;
         mPeerAddress = peerAddress;
+        mInstanceString = InstanceString;
+
         BluetoothSocket tmp = null;
         try {
             tmp = device.createInsecureRfcommSocketToServiceRecord(BtUUID);
```

Comment:
If this fails aren't we in a really bad place? This means that mSocket is going to be null forever so this object is essentially useless. In that case wouldn't it make more sense to just throw? Now, per the previous conversation, it doesn't always make sense to throw from a constructor. So I figure we could move the call to createInsecureRfcommSocketToServiceRecord to run() and then we can certainly throw from there. But one way or another if this function fails we need an exception somewhere.